src/packageurl/contrib/url2purl.py

```diff
@@ -79,6 +79,24 @@ def purl_from_pattern(type_, pattern, uri):
         return PackageURL(type_, **purl_data)
 
 
+def base_parsing(uri):
+    path = unquote_plus(urlparse(uri).path)
+    segments = [seg for seg in path.split('/') if seg]
+    
+    if len(segments)==1: 
+        return None
+
+    namespace = segments[0]
+    name = segments[1]
+
+    data = {
+        'namespace' : namespace,
+        'name' : name,
+        'segments' : segments
+    }
+    return data
+
+
 @purl_router.route('https?://registry.npmjs.*/.*',
                    'https?://registry.yarnpkg.com/.*')
 def build_npm_purl(uri):
```

Comment:
What about other URLs such https://www.npmjs.com/package/foo-utils ? or https://yarnpkg.com/package/foo ?

Reply:
@TG1999 any updates here?

Reply:
I have not changed logic for npm url parsing, I have just ran black for code formatting so it changed the formatting only

Reply:
Heh you are right. I have opened a ticket here as we will leave this out of scope for now: https://github.com/package-url/packageurl-python/issues/38